hazelcast/src/main/java/com/hazelcast/instance/impl/NodeExtension.java

```diff
@@ -176,6 +176,13 @@ default boolean isClientFailoverSupported() {
      */
     boolean isStartCompleted();
 
+    /**
+     * Check whether this instance is ready to start accepting requests. Use this
+     * method as a readiness probe.
+     * @return {@code true} when this instance is ready to accept requests, {@code false} otherwise.
+     */
+    boolean isReady();
```

Comment:
minor: it will be good if you can also add what steps should be completed to say that the node is ready to start accepting requests? like HR data should be loaded or partitions should be balanced.

Reply:
also worth to javaDoc difference between `isReady` and `isStartCompleted`